### DISCUSSION TITLE
Active scan selectively

### DISCUSSION THREAD

**Discussion by DTTerastar** [h2zero/NimBLE-Arduino#432] [Q&A] [answered]
Is it possible to ask for scan results for specific devices instead of every device?

**Comment by h2zero** [comment, marked as answer]
Active scan is a setting provided to the stack when starting a scan, so there is no way to do it selectively. The closest thing is starting and stopping with it enabled/disabled.

**Reply by DTTerastar** [reply]
Ok, that's what I thought.  You can't just send your own SCAN_REQs manually?

**Reply by h2zero** [reply]
Unfortunately no, it's all done at the controller level.

**Reply by Zimbu98** [reply]
I might be totally off target here, but I think DTTerastar might be asking if they could whitelist specific MACs and only scan for those. So my answer would be yes, this is possible. I think they might be using the words "active scan" by mistake. (there is a whitelist example in the Arduino examples for this library)

Additionally, they could scan for BLE things using the whitelist approach, then additionally send a scan request to that specific BLE item to get some more specific info. For example, I do this once a week to check the battery level on each beacon. My beacons will respond to a request to "2080" with the battery level. 

```
if( currentTime - previousBatteryCheckMillis >= dailyBatteryCheckIntervalMillis )
                      {
                        previousBatteryCheckMillis = currentTime;
                        if (advertisedDevice->getServiceData(NimBLEUUID("2080")) != "") 
                        {
                          batteryPower = (advertisedDevice->getServiceData<uint8_t>(NimBLEUUID("2080")));
                          Serial.print(". Battery power is: ");
                          Serial.print(batteryPower);
                          Serial.print("%");
                        }
                      }
```
